### PR TITLE
fix: only propagate FTL headers

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

File: internal/rpc/headers/headers.go [hunk 1]
```diff
@@ -24,6 +25,19 @@ const (
 	ParentRequestIDHeader = "Ftl-Parent-Request-Id"
 )
 
+var headers = []string{DirectRoutingHeader, VerbHeader, RequestIDHeader, ParentRequestIDHeader}
+
+// CopyRequest creates a new request with the same message as the original, but with only the FTL specific headers
+func CopyRequest[T any](req *connect.Request[T]) *connect.Request[T] {
```
Comment (matt2e) [inline internal/rpc/headers/headers.go:31]: Small nit, this sounds like a complete copy rather than what it is? Maybe `CopyRequestForForwarding`?

**Review by matt2e — APPROVED** [review]

File: backend/controller/controller.go [hunk 2]
```diff
@@ -853,7 +853,7 @@ func (s *Service) AcquireLease(ctx context.Context, stream *connect.BidiStream[f
 }
 
 func (s *Service) Call(ctx context.Context, req *connect.Request[ftlv1.CallRequest]) (*connect.Response[ftlv1.CallResponse], error) {
-	return s.callWithRequest(ctx, req, optional.None[model.RequestKey](), optional.None[model.RequestKey](), "")
+	return s.callWithRequest(ctx, headers.CopyRequest(req), optional.None[model.RequestKey](), optional.None[model.RequestKey](), "")
```
Comment (wesbillman) [inline backend/controller/controller.go:856]: Nice!

**Review by wesbillman — APPROVED** [review]

**Comment by alecthomas** [comment]
This isn't going to break OTEL tracing is it?

**Comment by stuartwdouglas** [comment]
Actually that is a good point, the logic should probably be to strip protocol level headers rather than just preserving known ones.

**Pull request merged**